examples/pipelines/kritis.yaml

```diff
@@ -5,51 +5,67 @@ metadata:
-          params:
-              - name: pathToDockerfile
-                value: deploy/Dockerfile
-          prevTasks: ['unit-test-kritis']
-        - name: deploy-test-env           # 3. Finally Deploy to Test environment
-          taskRef:
-              name: deploy-with-helm
-          sourceBindings:
-              - inputName: workspace
-                sourceKey: kritis
-          paramBindings:                 # Implicit dependency on buildPush task.
-              - inputName: testImage
-                taskName: buildPush
-                taskOutputName: builtImage
-          params:
-              - name: pathToHelmCharts
-                value: kritis-charts
-          clusterBindings:
-              - clusterName: test
-          nextTasks: ['integration-test']
-        - name: integration-test          # 4. Run Integration Tests in test cluster
-          taskRef:
-              name: integration-test-in-docker
-          sourceBindings:
-              - inputName: workspace
-                sourceKey: kritis
-          params:
-              - name: testArgs
-                value: "-e REMOTE_INTEGRATION=true"
+      - name: unit-test-kritis          # 1.  Run unit Tests
+        taskRef:
+          name: make
+        inputSourceBindings:
+          - name: workspace
+            sourceKey: kritis-app-github
+        params:
+          - name: makeTarget
+            type: test
+      - name: push-kritis               # 2.  Build And Push Tests
+        taskRef:
+          name: build-push
+        inputSourceBindings:
+          - name: workspace
+            sourceKey: kritis-app-github
+            passedConstraint: 
+              - make
+        outputSourceBindings:
+          - name: registry
+            sourceKey: stagingRegistry
+        params:
+          - name: pathToDockerfile
+            type: deploy/Dockerfile
+      - name: deploy-test-env           # 3. Finally Deploy to Test environment
+        taskRef:
+          name: deploy-with-helm
+        inputSourceBindings:
+          - name: registry
+            sourceKey: stagingRegistry
+            passedConstraint: [build-push]
+        params:
+          - name: pathToHelmCharts
+            type: kritis-charts
+        clusterBindings:
+          - clusterName: test
+      - name: integration-test          # 4. Run Integration Tests in test cluster
+        taskRef:
+          name: integration-test-in-docker
+        inputSourceBindings:
+          - name: workspace
+            sourceKey: kritis-test-github
+            passedConstraint: [deploy-with-helm]
+        params:
+          - name: testArgs
+            type: "-e REMOTE_INTEGRATION=true"
+    resources:
+      - name: kritis-app-github
+        resourceRef:
+          name: kritis-app-github
+        type: git
+        url: https://github.com/grafeas/kritis
+        revision: master
+      - name: kritis-test-github
+        resourceRef:
+          name: kritis-test-github
```

Comment:
where is this resourceRef defined?

Reply:
I think we might not need the `resourceRef`, will reconsider this and provide more details